celery/platforms.py

```diff
@@ -255,10 +255,16 @@ def create_pidlock(pidfile):
     return pidlock
 
 
+def _can_create_pidlock(pidlock):
```

Comment:
How about transferring this to `Pidfile` class? I would suggest something like `is_not_acquired` or similar. What do you think?

Reply:
I actually thought it would be better in the Pidfile class but I wanted to minimize changes.
I'll try to commit something soon.

Reply:
Shouldn't we change `Pidfile.is_locked()` instead to remove the pidfile if stale, and only then report on whether the lock is acquired? Something like:

```python
def is_locked(self):
    return os.path.exists(self.path) and not self.remove_if_stale()
```
It seems semantically more accurate and it is hard to imagine only wanting to check if the file exists without wanting to know if it is legitimately so. It would also save us from growing the `Pidfile` interface.

With this we can implement the helper function:

```python
def _free_or_die(pidfile):
    pidlock = Pidfile(pidfile)
    if pidlock.is_locked():
        print(PIDLOCKED.format(pidlock, pidlock.read_pid()), file=sys.stderr)
        raise SystemExit(EX_CANTCREAT)
    return pidlock
```

And reimplement the other 2 functions:

```python
def _create_pidlock(pidfile):
    return _free_pidfile_or_die(pidfile).acquire()

def after_chdir_do():
    ...
    if pidfile:
        _free_pidfile_or_die(pidfile)
```